binding/dotnet/PvSpeaker/PvSpeaker.cs

```diff
@@ -0,0 +1,441 @@
+        public static PvSpeaker Create(int sampleRate, int bitsPerSample, int bufferSizeSecs = 20, int deviceIndex = -1)
+        {
+            return new PvSpeaker(sampleRate, bitsPerSample, bufferSizeSecs, deviceIndex);
+        }
+
+        /// <summary>
+        /// Constructor for PvSpeaker.
+        /// </summary>
+        /// <param name="sampleRate">
+        /// The sample rate of the audio to be played.
+        /// </param>
+        /// <param name="bitsPerSample">
+        /// The number of bits per sample of the audio to be played.
+        /// </param>
+        /// <param name="bufferSizeSecs">
+        /// The size in seconds of the internal buffer used to buffer PCM data
+        /// - i.e. internal circular buffer will be of size `sampleRate` * `bufferSizeSecs`.
+        /// </param>
+        /// <param name="deviceIndex">
+        /// The index of the audio device to play audio from. A value of (-1) will use the default audio device.
+        /// </param>
+        private PvSpeaker(int sampleRate, int bitsPerSample, int bufferSizeSecs, int deviceIndex)
+        {
+            if (sampleRate <= 0)
+            {
+                throw new PvSpeakerInvalidArgumentException($"Sample rate of {sampleRate} is invalid - must be greater than 0.");
+            }
+
+            if (bitsPerSample != 8 && bitsPerSample != 16 && bitsPerSample != 24 && bitsPerSample != 32)
+            {
+                throw new PvSpeakerInvalidArgumentException($"Bits per sample of {bitsPerSample} is invalid - must be 8, 16, 24, or 32.");
+            }
+
+            if (bufferSizeSecs <= 0)
+            {
+                throw new PvSpeakerInvalidArgumentException($"Buffer size (in seconds) of {bufferSizeSecs} is invalid - must be greater than 0.");
+            }
+
+            if (deviceIndex < -1)
+            {
+                throw new PvSpeakerInvalidArgumentException($"Device index of {deviceIndex} is invalid - must be greater than -1.");
+            }
+
+            PvSpeakerStatus status = pv_speaker_init(sampleRate, bitsPerSample, bufferSizeSecs, deviceIndex, out _libraryPointer);
+            if (status != PvSpeakerStatus.SUCCESS)
+            {
+                throw PvSpeakerStatusToException(status);
+            }
+
+            SampleRate = sampleRate;
+            BitsPerSample = bitsPerSample;
+            BufferSizeSecs = bufferSizeSecs;
+            SelectedDevice = Marshal.PtrToStringAnsi(pv_speaker_get_selected_device(_libraryPointer));
+            Version = Marshal.PtrToStringAnsi(pv_speaker_version());
+        }
+
+        /// <summary>
+        /// Starts the audio output device. Should be called before making any calls to `Write()`, `Flush()`, or `Stop()`.
+        /// </summary>
+        public void Start()
+        {
+            PvSpeakerStatus status = pv_speaker_start(_libraryPointer);
+            if (status != PvSpeakerStatus.SUCCESS)
+            {
+                throw PvSpeakerStatusToException(status);
```

Comment:
might be nice to add a message to each failure point. In this case, something like `Failed to start PvSpeaker.`